repository/src/main/java/org/alfresco/repo/search/impl/querymodel/impl/db/NodePermissionAssessor.java

```diff
@@ -0,0 +1,228 @@
+import java.io.Serializable;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+import org.alfresco.model.ContentModel;
+import org.alfresco.repo.cache.lookup.EntityLookupCache;
+import org.alfresco.repo.domain.node.Node;
+import org.alfresco.repo.domain.permissions.Authority;
+import org.alfresco.repo.security.authentication.AuthenticationUtil;
+import org.alfresco.service.cmr.repository.NodeRef;
+import org.alfresco.service.cmr.repository.NodeService;
+import org.alfresco.service.cmr.repository.datatype.DefaultTypeConverter;
+import org.alfresco.service.cmr.security.PermissionService;
+import org.alfresco.service.namespace.QName;
+import org.alfresco.util.EqualsHelper;
+
+public class NodePermissionAssessor
+{
+    private final boolean isSystemReading;
+    private final boolean isAdminReading;
+    private final boolean isNullReading;
+    private final Authority authority;
+    private final Map<Long, Boolean> aclReadCache = new HashMap<>();
+    private int checksPerformed;
+    private long startTime;
+    private int maxPermissionChecks;
+    private long maxPermissionCheckTimeMillis;
+    
+    private EntityLookupCache<Long, Node, NodeRef> nodesCache;
+    private NodeService nodeService;
+    private PermissionService permissionService;
+
+    public NodePermissionAssessor(NodeService nodeService, PermissionService permissionService,
+            Authority authority, EntityLookupCache<Long, Node, NodeRef> nodeCache)
+    {
+        this.permissionService = permissionService;
+        this.nodesCache = nodeCache;
+        this.nodeService = nodeService;
+        
+        this.checksPerformed = 0;
+        this.maxPermissionChecks = Integer.MAX_VALUE;
+        this.maxPermissionCheckTimeMillis = Long.MAX_VALUE;
+
+        Set<String> authorisations = permissionService.getAuthorisations();
+        this.isSystemReading = AuthenticationUtil.isRunAsUserTheSystemUser();
+        this.isAdminReading = authorisations.contains(AuthenticationUtil.getAdminRoleName());
+        this.isNullReading = AuthenticationUtil.getRunAsUser() == null;
+
+        this.authority = authority;
+    }
+
+    public boolean isIncluded(Node node)
+    { 
+        if (isFirstRecord())
+        {
+            this.startTime = System.currentTimeMillis();
+        }
+        
+        checksPerformed++;
+        return isReallyIncluded(node);
+    }
+
+    public boolean isFirstRecord()
+    {
+        return checksPerformed == 0;
+    }
+
+    protected boolean isOwnerReading(Node node, Authority authority)
+    {
+        aclOwnerStopWatch().start();
+        try
+        {
+            if (authority == null)
+            {
+                return false;
+            }
+            
+            String owner = getOwner(node);
+            if (EqualsHelper.nullSafeEquals(authority.getAuthority(), owner))
+            {
+                return true;
+            }
+            else
+            {
+                return false;
+            }
```

Comment:
you could just return `EqualsHelper.nullSafeEquals(authority.getAuthority(), owner)`

Reply:
You're right, now the method is returning  `EqualsHelper.nullSafeEquals(authority.getAuthority(), owner)` for that specific code block.